components/fxa-client/android/src/main/java/mozilla/appservices/fxaclient/FirefoxAccount.kt

```diff
@@ -225,6 +225,127 @@ class FirefoxAccount(handle: FxaHandle, persistCallback: PersistCallback?) : Aut
         }.getAndConsumeRustString()
     }
 
+    /**
+     * Update the push subscription details for the current device.
+     * This needs to be called every time a push subscription is modified or expires.
+     *
+     * This performs network requests, and should not be used on the main thread.
+     *
+     * @param endpoint Push callback URL
+     * @param endpoint Public key used to encrypt push payloads
+     * @param endpoint Auth key used to encrypt push payloads
+     */
+    fun setDevicePushSubscription(endpoint: String, publicKey: String, authKey: String) {
+        rustCall { e ->
+            LibFxAFFI.INSTANCE.fxa_set_push_subscription(this.handle.get(), endpoint, publicKey, authKey, e)
+        }
+    }
+
+    /**
+     * Update the display name (as shown in the FxA device manager, or the Send Tab target list)
+     * for the current device.
+     *
+     * This performs network requests, and should not be used on the main thread.
+     *
+     * @param displayName The current device display name
+     */
+    fun setDeviceDisplayName(displayName: String) {
+        rustCall { e ->
+            LibFxAFFI.INSTANCE.fxa_set_device_name(this.handle.get(), displayName, e)
+        }
+    }
+
+    /**
+     * Retrieves the list of the connected devices in the current account, including the current one.
+     *
+     * This performs network requests, and should not be used on the main thread.
+     */
+    fun getDevices(): Array<Device> {
```

Comment:
Currently, consuming application is expected to cache this, right? Afaik every call to `getDevices` will hit the network currently. That's probably fine for the first iteration. Keeping a cache of devices updated will be an annoyance...

Reply:
FWIW it seems reasonable for this method to grow its own internal caching over time, especially since we're supposed to receive push notifications that tell us when to invalidate that cache.